ban420 homework 4/timer.r Solution 3

```diff
@@ -0,0 +1,35 @@
+library(foreach)
+library(doParallel)
+library(tictoc)
+
+MTweedieTests <- function(N, M, sig) {
+  tic("MTweedieTests Timing")
+  
+  numCores <- 8  
+  cl <- makeCluster(numCores)
+  registerDoParallel(cl)
+  
+  clusterExport(cl, varlist = c("simTweedieTest"))
+  clusterEvalQ(cl, library(tweedie))
+
+  p.values <- foreach(i = 1:M, .combine = c) %dopar% {
+    simTweedieTest(N)
+  }
+  
+  share_reject <- mean(p.values < sig)
+  
+  stopCluster(cl)
+  
+  toc()
+  
+  return(share_reject)
+}
+
+
+
+# Run the MTweedieTests function
+result <- MTweedieTests(N = 1000, M = 1000, sig = 0.05)
```

Comment:
This is not directly comparable to method 1 and 2, as you only run 1000 simulations in this case and 5000 (5 * 1000) simulations in the other methods. You have however done it correctly.